### PR TITLE
76 one page capmaign

### LINKED ISSUE #76: One Page Campaign
### Description
Create campaign page that allow user to see full information about this campaign.

* Route:	``'/campaign/id'``
* Signed in only:	`flase`
* Admin only:	`false`

### Component:
- Donation Form

### Screenshot:
image

### PR DESCRIPTION
**Pull request by ghost**
Result: 
image
image

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -32,7 +33,7 @@ function App() {
         <Route
           path="/campaigns"
           element={
-            <h1>campaigns</h1>
+            <OneCampaign id={2} />
```
Comment (HaniOlwan) [inline client/src/App.js:36]: i think in the campaign itself you will make a get request to /api/campagin/${id}

File: server/controllers/compaignRouter/getCampaignById.ts [hunk 2]
```diff
@@ -1,16 +1,68 @@
...
+
+    const numOfFamilies = await Capon.findAll({
+      where: {
+        campaignId: id,
+      },
+      raw: true,
+    });
     if (!campaignInfo) throw new CustomedError('There is no campaign', 400);
-    res.status(200).json({ message: 'Success', data: campaignInfo });
+    res.status(200).json({
```
Comment (Raghad-Mezied) [inline server/controllers/compaignRouter/getCampaignById.ts:58]: You don't need to send the status code of 200 as it is the default

File: server/controllers/compaignRouter/getCampaignById.ts [hunk 3]
```diff
@@ -1,16 +1,68 @@
...
+        ],
+      ],
+    });
+
+    const numOfFamilies = await Capon.findAll({
+      where: {
+        campaignId: id,
+      },
+      raw: true,
+    });
```
Comment (Raghad-Mezied) [inline server/controllers/compaignRouter/getCampaignById.ts:56]: As all these don't depend on each other you can use PromiseAll.
And you here adding the clothes, money, and food together? they are all different scales.

File: client/src/pages/OneCampaign/index.jsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+} from 'antd';
+import axios from 'axios';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import './style.less';
+import { TeamOutlined } from '@ant-design/icons';
+import { DonationButton } from '../../components';
+
+const { Title, Paragraph } = Typography;
+function OneCampaign({ id }) {
```
Comment (Raghad-Mezied) [inline client/src/pages/OneCampaign/index.jsx:19]: Call it just Campaign, and you don't need to pass the id here as you can access it through the params

File: client/src/pages/OneCampaign/index.jsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      try {
+        const {
+          data: { data },
+        } = await axios.get(`/api/campaign/${id}`, {
+          cancelToken: source.token,
+        });
+        setCampaign(data.campaignInfo);
+        setCurrent(data.current);
+        setTarget(data.campaignInfo.target);
+        setFamilies(data.families);
```
Comment (Raghad-Mezied) [inline client/src/pages/OneCampaign/index.jsx:37]: Why did you split them into different states?

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by farah2003 — APPROVED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by DinaAlghazali — APPROVED** [review]

**Review by haniolwan — APPROVED** [review]

**Pull request merged**